app/api/categories/route.ts

```diff
@@ -0,0 +1,33 @@
+import { NextResponse } from 'next/server';
+
+import { getCategoryTree } from '~/client/queries/get-category-tree';
+
+interface Category {
+  entityId: number;
+  name: string;
+  path: string;
+}
+
+type CategoryWithChildren = Category & { children?: CategoryWithChildren[] };
+
+const flattenCategoryTree = (categories: CategoryWithChildren[]) => {
+  return categories.reduce<Category[]>((acc, category) => {
+    acc.push(category);
+
+    if (category.children?.length) {
+      acc.push(...flattenCategoryTree(category.children));
+    }
+
+    return acc;
+  }, []);
```

Comment:
I feel like every time I have tried adding a type to reduce I have had to cast my empty initial array. 🤔 

```diff
- }, []);
+ }, [] as Category[]);
```

Glad you didn't have to.

Reply:
I use the `.reduce<Category[]>` generic to type it instead